File: rb/features.rb
# PokemonTcg SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module PokemonTcgFeatures
  def self.make_feature(name)
    case name
    when "base"
      PokemonTcgBaseFeature.new
    when "test"
      PokemonTcgTestFeature.new
    else
      PokemonTcgBaseFeature.new
    end
  end
end

File: py/pokemontcg_sdk/features.py
# PokemonTcg SDK feature factory

from pokemontcg_sdk.feature.base_feature import PokemonTcgBaseFeature
from pokemontcg_sdk.feature.test_feature import PokemonTcgTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PokemonTcgBaseFeature(),
        "test": lambda: PokemonTcgTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
